streams/src/main/java/org/apache/kafka/streams/processor/assignment/assignors/StickyTaskAssignor.java

```diff
@@ -344,7 +341,7 @@ public ProcessId findLeastLoadedClient(final TaskId taskId, final Set<ProcessId>
                 }
 
                 if (leastLoaded == null || thisClientLoad < clientLoad(leastLoaded)) {
-                    final Set<TaskId> assignedTasks = newAssignments.getOrDefault(processId, new HashSet<>())
+                    final Set<TaskId> assignedTasks = newAssignments.getOrDefault(processId, KafkaStreamsAssignment.of(processId, new HashSet<>())).tasks().values()
```

Comment:
See above, would be nice to clean up some of the getOrDefaults if possible (but it's not a huge deal, they don't add that much complexity)